pkgs/build-support/rust/build-rust-package/default.nix

```diff
@@ -44,6 +44,11 @@
 , useNextest ? false
 , depsExtraArgs ? {}
 
+# You must use "--offline" instead of "--frozen" if `Cargo.lock` is
+# valid-but-outdated, for example when adding a `[patch]` stanza to
+# `Cargo.toml` without regenerating `Cargo.lock`.
```

Comment:
If you're adding a patch stanza why can't you regenerate the lock at the same time?

Reply:
Because you're adding the patch stanza from inside a derivation!

This lets us do for crate dependencies of a `buildRustPackage` what we can already do for ordinary packages in nixpkgs -- override a specific dependency throughout all of nixpkgs.

My own motivating example case is the `ring` crate, which is basically "openssl for Rust".  It's a deep, deep dependency of a huge chunk of the ecosystem.  Ring does not build on PowerPC, and the Ring maintainer has ignored a patch submitted by IBM *for over two full years.*  There needs to be a way for people to route around this sort of nonsense.

But this is generally useful, for the same reasons why `.override` is useful.

Reply:
Here is the example use:

https://github.com/NixOS/nixpkgs/commit/78b4b5bf68d41f412944a2ee846715dfb090d045

Reply:
> This lets us do for crate dependencies of a buildRustPackage what we can already do for ordinary packages in nixpkgs -- override a specific dependency throughout all of nixpkgs.

How does that work?

Reply:
For the example you posted you can do this:

> Patch toml and lock, add to cargoPatches.